lib/controllers/DeviceController.ts

```diff
@@ -82,20 +88,38 @@ export class DeviceController extends CRUDController {
     const engineId = request.getString('engineId');
     const model = request.getBodyString('model');
     const reference = request.getBodyString('reference');
-    const assetId = request.input.args.assetId || null;
     const metadata = request.getBodyObject('metadata', {});
     const refresh = request.getRefresh();
 
+    let assetId = null;
+    try {
+      assetId = request.getBodyString('assetId');
+    }
+    catch (error) {}
+    const measureNamesLinks = request.getBodyArray('measureNamesLinks', []);
+
+    if (measureNamesLinks.length && ! this.validateMeasureNamesLinks(measureNamesLinks)) {
+      throw new PluginImplementationError('The linkRequest provided is incorrectly formed');
+    }
+
     const deviceContent: DeviceContent = {
       measures: [],
-      measuresName: [],
       metadata,
       model,
       reference,
     };
 
-    const device = await this.deviceService.create(engineId, deviceContent, assetId, {
-      refresh
+    const linkRequest: LinkRequest = assetId && measureNamesLinks.length
+      ? {
+        assetId,
+        deviceLink: { deviceId: Device.id(model, reference), measureNamesLinks }
+      }
+      : null;
```

Comment:
remove the check.

Reply:
Why was that? Was it  for the length of the array `measureNamesLinks` I introduced?